live_test.go

```diff
@@ -0,0 +1,110 @@
+package gollm
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLiveAPICallWithGenerate(t *testing.T) {
```

Comment:
I added this "live" test to verify that I hadn't broken openai response parsing. I added an ollama test as well. I don't have keys to other providers, but those that do could add similar Runners.

Note, the tests are skipped if the API keys (for remote services like openai) or valid endpoints (like Ollama) are not present.